File: Unity Masters - Prototype 4 - Level 2/Assets/GenerateObjects.js
#pragma strict
var amount : int;
var prefab : Transform;
var rand : Random;

function Start () {

}

function Update () {

}

function Generate(){

	var objects = GameObject.FindGameObjectsWithTag("2");
	for(var go : GameObject in objects){
		Destroy(go);
	}
	
	for(var i=0;i<amount;i++){
		var x  = Random.Range(-10,10);
		var y = Random.Range(-22,22);
	 	Instantiate(prefab,Vector3(x,2,y) ,Quaternion.identity);
	 }
}

File: Level two/Assets/GenerateObjects.cs
using UnityEngine;
using System.Collections;

public class GenerateObjects : MonoBehaviour {
	
	Transform prefab;
	Random r = new Random();
	
	
	// Use this for initialization
	void Start () {
		prefab = GameObject.Find("Collect").transform;
		Generate(5);
	
	}
	
	void Generate(int num){
		for(int i =0; i<num; i++){
			float x = UnityEngine.Random.Range(-22,22);
			float y = UnityEngine.Random.Range(-22,22);
			Debug.Log("x" +x);
			Debug.Log("y" +y);
			
			Debug.Log("Testing" + prefab.name);

			Transform g = Instantiate(prefab, new Vector3(x,y,0),Quaternion.identity) as Transform;
		}
	}
	
	// Update is called once per frame
	void Update () {
	
	}
}
